lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -111,16 +123,72 @@ define((require, exports, module) ->
         next: popState
       }
 
+    mathStartRule = (openingRegex, state) -> {
+      token: "string"
+      regex: openingRegex
+      next: pushState(state)
+      merge: false
+    }
+
+    mathEndRules = (closingRegex) -> [
+      { token: "string", regex: closingRegex, next: popState }
+      { token: "error", regex : "^\\s*$", next: popState }
+    ]
+
     specificTokenForState = {}
-    specificTokenForState[LIST_STATE] = LIST_TOKENTYPE
-    specificTokenForState[EQUATION_STATE] = EQUATION_TOKENTYPE
-    specificTokenForState[MATH_STATE] = EQUATION_TOKENTYPE
-    specificTokenForState[MATH_LATEX_STATE] = EQUATION_TOKENTYPE
+    specificTokenForState[LIST_ITEMIZE_STATE] = LIST_TOKENTYPE
+    specificTokenForState[LIST_ENUMERATE_STATE] = LIST_TOKENTYPE
+    specificTokenForState[EQUATION_REGULAR_STATE] = EQUATION_TOKENTYPE
+    specificTokenForState[EQUATION_ASTERISK_STATE] = EQUATION_TOKENTYPE
+    specificTokenForState[MATH_INLINE_STATE] = EQUATION_TOKENTYPE
+    specificTokenForState[MATH_MULTILINE_STATE] = EQUATION_TOKENTYPE
+    specificTokenForState[MATH_ALTERNATIVE_STATE] = EQUATION_TOKENTYPE
+
+    equationStartRules = [
+      beginRule(EQUATION_REGULAR_REGEX, EQUATION_REGULAR_STATE)
+      beginRule(EQUATION_ASTERISK_REGEX, EQUATION_ASTERISK_STATE)
+      mathStartRule(MATH_MULTILINE_OPENING_REGEX, MATH_MULTILINE_STATE)
+      mathStartRule(MATH_INLINE_OPENING_REGEX, MATH_INLINE_STATE)
+      mathStartRule(MATH_ALTERNATIVE_OPENING_REGEX, MATH_ALTERNATIVE_STATE)
+    ]
+
+    citationsRules = [
+      {
+        token: [
+          "storage.type"
+          "lparen.ref"
+          "variable.parameter.ref"
+          "rparen"
+        ]
+        regex: "(\\\\(?:ref))({)(\\w*)(})"
+      }
+      {
+        token: [
+          "storage.type"
+          "lparen.cite"
+          "variable.parameter.cite"
+          "rparen"
+        ]
+        regex: "(\\\\(?:cite))({)(\\w*)(})"
+      }
+      {
+        token: [
+          "keyword"
+          "lparen"
+          "variable.parameter"
+          "rparen"
+        ]
+        regex: "(\\\\(?:v?ref|cite(?:[^{]*)))(?:({)([^}]*)(}))?"
```

Comment:
can you add a comment like "this handles \vref and \vcite macros" ? Otherwise the difference between this regexp and two previous ones is not obvious